edk2toollib/windows/telem/cper_parser_tool.py

```diff
@@ -0,0 +1,659 @@
+        return str(self.persistenceInfo)
+
+    def PrettyPrint(self) -> str:
+        '''Print relevant portions of the CPER header. Change to suit your needs'''
+
+        string = ""
+        temp = ""
+
+        string += "Record Length:  " + str(self.recordLength) + '\n'
+
+        temp = self.TimestampParse()
+        string += "Time of error:  " + temp + '\n'
+
+        temp = self.ErrorSeverityParse()
+        string += "Error severity: " + temp + '\n'
+
+        temp = self.FlagsParse()
+        string += "Flags Id:       " + temp + '\n'
+
+        temp = self.PlatformIdParse()
+        string += "Platform Id:    " + temp + '\n'
+
+        temp = self.PartitionIdParse()
+        string += "Partition Id:   " + temp + '\n'
+
+        temp = self.CreatorIdParse()
+        string += "Creator Id:     " + temp + '\n'
+
+        return string[0:-1]  # omit the last newline
+
+
+class CPER_SECTION_HEADER(object):
+    '''TODO: Fill in'''
+
+    STRUCT_FORMAT = "=IIHccI16s16sI20s"
+
+    def __init__(self, input: str):
+
+        self.FlagList = []  # go to self.FlagsParse() to see description of this field
+        self.fruIdValid = False
+        self.fruStringValid = False
+
+        (self.sectionOffset,
+            self.sectionLength,
+            self.revision,
+            self.validationBits,
+            self.reserved,
+            self.flags,
+            self.sectionType,
+            self.fruId,
+            self.sectionSeverity,
+            self.fruString) = struct.unpack_from(self.STRUCT_FORMAT, input)
+
+        self.FlagsParse()
+        self.ValidBitsParse()
+
+    def RevisionParse(self) -> str:
+        '''
+        Parse the major and minor version number of the Error Record definition
+
+        TODO: Actually parse the zeroth and first byte which represent the minor
+        and major version numbers respectively
+        '''
+
+        return str(self.revision)
+
+    def ValidBitsParse(self) -> None:
+        '''
+        if bit 0: FruId contains valid info\n
+        if bit 1: FruId String contains valid info
+        '''
+
+        if(ord(self.validationBits) & 0b1):  # check bit 0
+            self.fruIdValid = True
+        if(ord(self.validationBits) & 0b10):  # check bit 1
+            self.fruStringValid = True
+
+    def FlagsParse(self) -> None:
+        '''
+        Check the flags field and populate list containing applicable flags
+
+        if bit 0: This is the section to be associated with the error condition\n
+        if bit 1: The error was not contained within the processor or memery heirarchy\n
+        if bit 2: The component has been reset and must be reinitialized\n
+        if bit 3: Error threshold exceeded for this component\n
+        if bit 4: Resource could not be queried for additional information\n
+        if bit 5: Action has been taken to contain the error, but the error has not been corrected
+        '''
+
+        FlagList = []
+
+        if(self.flags & int('0b1', 2)):  # Check bit 0
```

Comment:
you do this a lot, maybe move this to a helper function, ie check_bit for example? Just a suggestion.